### PR TITLE
Fix the qemu cmd line change in rhel9

### PR DESCRIPTION
**Pull request by jinqi7**
Signed-off-by: qijing <jinqi@redhat.com>

Fix the qemu cmd line change related to memory in rhel9

### CODE REVIEW

**Comment by jinqi7** [comment]
# avocado run --vt-type libvirt virsh.migrate_mem.mem_device
WARNING:root:No python imaging library installed. Screendump and Windows guest BSOD detection are disabled. In order to enable it, please install python-imaging or the equivalent for your distro.
WARNING:root:No python imaging library installed. PPM image conversion to JPEG disabled. In order to enable it, please install python-imaging or the equivalent for your distro.
JOB ID     : f79a37c996dcd18d179d19cfac11c372b7a7adbc
JOB LOG    : /root/avocado/job-results/job-2021-07-14T05.04-f79a37c/job.log
 (1/1) type_specific.io-github-autotest-libvirt.virsh.migrate_mem.mem_device: PASS (194.58 s)
RESULTS    : PASS 1 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB TIME   : 197.57 s

**Review by chunfuwen — APPROVED** [review]
LGTM

File: libvirt/tests/cfg/migration/migrate_mem.cfg [hunk 1]
```diff
@@ -49,7 +49,7 @@
             memdev_tg_sizeunit = "MiB"
             memdev_tg_node = "0"
             # migrate from 8.2 to 8.3.1, the qemu cmdline is "-numa node,nodeid=1,cpus=2-3,mem=1024 -object memory-backend-ram,id=memdimm0,size=536870912"
-            qemu_checks = "(id=ram-node1,size=536870912|)`-numa node,nodeid=1,cpus=2-3,mem(=1024|dev=ram-node1)`-object memory-backend-ram,id=memdimm0,size=536870912"
+            qemu_checks = "?memory-backend-ram"?,"?id"?(=|:)"?memdimm0"?,"?size"?(=|:)"?536870912"?
```
Comment (Yingshun) [inline libvirt/tests/cfg/migration/migrate_mem.cfg:52]: `(=|:)` should be `[=:]`, r8?

Comment (jinqi7) [reply libvirt/tests/cfg/migration/migrate_mem.cfg:52]: Both can work.  What the expectation here is It can be found.

Comment (Yingshun) [reply libvirt/tests/cfg/migration/migrate_mem.cfg:52]: Ok, got it.

**Review by Yingshun — APPROVED** [review]

**Review by dzhengfy — APPROVED** [review]
LGTM

**Pull request merged**